examples/joy_circle_grid.ml

```diff
@@ -0,0 +1,34 @@
+open Joy.Shape
+
+let grid_of_circles canvas_width canvas_height spacing =
+  let create_circle x y = circle 50 |> translate x y in
+  let num_circles_x = canvas_width / spacing in
+  let num_circles_y = canvas_height / spacing in
+  let initial_x = (canvas_width - (num_circles_x * spacing)) / 2 in
+  let initial_y = (canvas_height - (num_circles_y * spacing)) / 2 in
+
+  let circles =
+    List.flatten
+      (List.map
+         (fun i ->
+           List.map
+             (fun j ->
+               create_circle
+                 (initial_x + (i * spacing))
+                 (initial_y + (j * spacing)))
+             (List.init num_circles_y (fun x -> x)))
+         (List.init num_circles_x (fun x -> x)))
```

Comment:
This can be simplified.

Because `List.init` already gives us an index, we can directly start creating circles with it.

I mean something like this:

```ocaml
let circles =
  List.flatten (List.init num_circles_x (fun i ->
    List.init num_circles_y (fun j ->
      let x_position = initial_x + (i * spacing) in
      let y_position = initial_y + (j * spacing) in
      create_circle x_position y_position
    )
  ))
```

Reply:
Absolutely! I've gone through the code you provided and I can see that is better structured and more simplified. I'm sorry for making it this complex. I was in a bit of a hurry because I had given myself a deadline to submit my first pr and I didn't want to miss it even though I was not feeling well. I am better now so please let me know what you would have me do now. If I am to correct some things or just start over. Thank you!

Reply:
You can make changes and push them in the same branch. That will update your PR.

Ideally, you should work on a branch with some name other than `main`, because that should be in sync with the original branch and not have un-merged changes. https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project

Reply:
Hi @nikochiko I'd be on it and make changes asap

Reply:
@TitilayoAdeyemi i don't see another pull request from you, other than `CI for Joy`.

Once you create a pull request, can you please link it here?